src/fragments/lib/datastore/ios/advanced-workflows/custom-primary-key/query-snippet.mdx

```diff
@@ -1 +1,73 @@
-## TODO (see js snippet for examples)
\ No newline at end of file
+A record with a custom primary key can be queried for in the following ways:
+
+With the value of the primary key:
+
+<BlockSwitcher>
+
+<Block name="Listener (iOS 11+)">
+
+```swift
+Amplify.DataStore.query(Book.self, byIdentifier: .identifier(isbn: "12345")) { result in 
+    switch result {
+    case .success(let book):
+        print("Query was successful, retrieved book: \(book)")
+    case .failure(let error):
+        print("Error on query() for type Book with error \(error)")
+    }
+}
+```
+
+</Block>
+
+<Block name="Combine (iOS 13+)">
+
+```swift
+let sink = Amplify.DataStore.query(Book.self, byIdentifier: .identifier(isbn: "12345")).sink {
+    if case let .failure(error) = $0 {
+        print("Error on query() for type Book with error \(error)")
+    }
+}
+receiveValue: { result in
+    print("Query was successful, retrieved book: \(result)")
```

Comment:
since `result` is an optional result for queries, let's get rid of the warning and make it clear by unwrapping it
```swift
guard let book = book else {
    print("Query was successful, no results.")
    return
}
            
print("Query was successful, retrieved book: \(book)")
```